File: debot/src/androidTest/java/com/tomoima/debot/DebotTestUtils.java
package com.tomoima.debot;


import android.app.Activity;
import android.support.v4.app.FragmentManager;

import java.util.concurrent.CountDownLatch;

import static android.support.test.InstrumentationRegistry.getInstrumentation;

class DebotTestUtils {

    static TestDebot getTestDebot(Activity activity){
        TestDebot debot = TestDebot.getInstance();
        getInstrumentation().waitForIdleSync();
        try {
            waitForFragmentTransaction(activity);
            return debot;
        } catch (InterruptedException e) {
            e.printStackTrace();
            return new TestDebot();
        }
    }

    private static void waitForFragmentTransaction(final Activity activity) throws InterruptedException {
        final CountDownLatch latch = new CountDownLatch(1);
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                activity.getFragmentManager().executePendingTransactions();
                latch.countDown();
            }
        });
        latch.await();
    }

    static class TestDebot extends DebotDialog {

        public static TestDebot getInstance() {
            return new TestDebot();
        }

        @Override
        public void show(FragmentManager manager, String tag) {

        }

    }

}

File: debot/src/androidTest/java/com/tomoima/debot/DebotTestUtils.kt
package com.tomoima.debot


import android.app.Activity
import android.app.Dialog
import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
import org.mockito.Mockito.*
import java.util.concurrent.CountDownLatch

internal object DebotTestUtils {

    fun getTestDebot(activity: Activity): TestDebot {
        val debot = TestDebot.instance
        getInstrumentation().waitForIdleSync()
        try {
            waitForFragmentTransaction(activity)
            return debot
        } catch (e: InterruptedException) {
            e.printStackTrace()
            return TestDebot()
        }

    }

    @Throws(InterruptedException::class)
    private fun waitForFragmentTransaction(activity: Activity) {
        val latch = CountDownLatch(1)
        activity.runOnUiThread {
            activity.fragmentManager.executePendingTransactions()
            latch.countDown()
        }
        latch.await()
    }

    class TestDebot : DebotDialog() {

        companion object {
            val instance: TestDebot
                get() = TestDebot()
        }

        override fun getDialog(): Dialog {
            return mock(Dialog::class.java);
        }

    }

}
